main.tf

```
@@ -126,8 +140,11 @@ resource "aws_elasticsearch_domain" "default" {
     dedicated_master_type    = var.dedicated_master_type
     zone_awareness_enabled   = var.zone_awareness_enabled
 
-    zone_awareness_config {
-      availability_zone_count = var.availability_zone_count
+    dynamic "zone_awareness_config" {
+      for_each = null_resource.azs[*].triggers
```

```suggestion
      for_each = var.availability_zone_count > 1 ? ["true"] : []
```

it could be done w/o using `null_resource`